src/pybind/mgr/dashboard/frontend/src/app/ceph/nfs/nfs-form/nfs-form.component.ts

```diff
@@ -127,12 +131,22 @@ export class NfsFormComponent extends CdForm implements OnInit {
 
     if (this.isEdit) {
       this.action = this.actionLabels.EDIT;
-      this.route.params.subscribe((params: { cluster_id: string; export_id: string }) => {
-        this.cluster_id = decodeURIComponent(params.cluster_id);
-        this.export_id = decodeURIComponent(params.export_id);
-        promises.push(this.nfsService.get(this.cluster_id, this.export_id));
-        this.getData(promises);
-      });
+      this.route.params.subscribe(
+        (params: { cluster_id: string; export_id: string; rgw_export_type?: string }) => {
+          this.cluster_id = decodeURIComponent(params.cluster_id);
+          this.export_id = decodeURIComponent(params.export_id);
+          if (params.rgw_export_type) {
+            this.nfsForm.get('rgw_export_type').setValue(params.rgw_export_type);
+            if (params.rgw_export_type === 'bucket') {
+              this.resolveBucket();
+            } else {
+              this.resolveUsers();
+            }
```

Comment:
what about ?
```suggestion
            this.nfsForm.get('rgw_export_type').setValue(params.rgw_export_type);
            if (params.rgw_export_type === 'bucket') {
              this.resolveBucket();
            } else {
              this.resolveUsers();
            }
```

Reply:
We need if condition of rgw_export_type here as this only applies in case of RGW export and not CEPHFS based exports & unit-test already failed just bcz of this reason 
```
NfsFormComponent
    ✓ should create (446 ms)
    ✓ should create the form (25 ms)
    ✓ should prepare data when selecting an cluster (22 ms)
    ✓ should not allow changing cluster in edit mode (24 ms)
    ✓ should mark NFSv4 & NFSv3 protocols as enabled always (22 ms)
    ✕ should match backend squash values with ui values (27 ms)
    should submit request
      ✓ should call update (28 ms)
      ✓ should call update with selected nfs protocol (23 ms)
      ✓ should call create (27 ms)
    pathExistence
      ✓ path cannot be empty (22 ms)
      ✓ path that does not exist should be invalid (17 ms)
      ✓ path that exists should be valid (19 ms)

  ● NfsFormComponent › should match backend squash values with ui values

    Expected no open requests, found 1: GET api/rgw/user?daemon_name=daemon1

      59 |       squash: backendSquashValue
      60 |     });
    > 61 |     httpTesting.verify();
         |                 ^
      62 |     expect(component.nfsForm.value).toMatchObject({
      63 |       squash: uiSquashValue
      64 |     });

      at HttpClientTestingBackend.verify (node_modules/@angular/common/fesm2020/http/testing.mjs:295:19)
      at matchSquash (src/app/ceph/nfs/nfs-form/nfs-form.component.spec.ts:61:17)
      at src/app/ceph/nfs/nfs-form/nfs-form.component.spec.ts:130:5
      at _ZoneDelegate.Object.<anonymous>._ZoneDelegate.invoke (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:409:30)
      at ProxyZoneSpec.Object.<anonymous>.ProxyZoneSpec.onInvoke (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:3830:43)
      at _ZoneDelegate.Object.<anonymous>._ZoneDelegate.invoke (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:408:56)
      at Zone.Object.<anonymous>.Zone.run (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:169:47)
      at Object.wrappedFunc (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:4330:34)
      ```

Reply:
We need if condition of rgw_export_type here as this only applies in case of RGW export and not CEPHFS based exports & unit-test already failed just bcz of this reason 
```
NfsFormComponent
    ✓ should create (446 ms)
    ✓ should create the form (25 ms)
    ✓ should prepare data when selecting an cluster (22 ms)
    ✓ should not allow changing cluster in edit mode (24 ms)
    ✓ should mark NFSv4 & NFSv3 protocols as enabled always (22 ms)
    ✕ should match backend squash values with ui values (27 ms)
    should submit request
      ✓ should call update (28 ms)
      ✓ should call update with selected nfs protocol (23 ms)
      ✓ should call create (27 ms)
    pathExistence
      ✓ path cannot be empty (22 ms)
      ✓ path that does not exist should be invalid (17 ms)
      ✓ path that exists should be valid (19 ms)

  ● NfsFormComponent › should match backend squash values with ui values

    Expected no open requests, found 1: GET api/rgw/user?daemon_name=daemon1

      59 |       squash: backendSquashValue
      60 |     });
    > 61 |     httpTesting.verify();
         |                 ^
      62 |     expect(component.nfsForm.value).toMatchObject({
      63 |       squash: uiSquashValue
      64 |     });

      at HttpClientTestingBackend.verify (node_modules/@angular/common/fesm2020/http/testing.mjs:295:19)
      at matchSquash (src/app/ceph/nfs/nfs-form/nfs-form.component.spec.ts:61:17)
      at src/app/ceph/nfs/nfs-form/nfs-form.component.spec.ts:130:5
      at _ZoneDelegate.Object.<anonymous>._ZoneDelegate.invoke (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:409:30)
      at ProxyZoneSpec.Object.<anonymous>.ProxyZoneSpec.onInvoke (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:3830:43)
      at _ZoneDelegate.Object.<anonymous>._ZoneDelegate.invoke (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:408:56)
      at Zone.Object.<anonymous>.Zone.run (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:169:47)
      at Object.wrappedFunc (node_modules/zone.js/bundles/zone-testing-bundle.umd.js:4330:34)
      ```

Reply:
Okay, thanks for explaining, I think for most of the places in this form we are hitting storage backend / FSAL check so lets add backend check wherever possible, so as to seperate interests.

Additionally,
Lets create a separate function for resolving route and handle all validations there.
I see CREATE is also subscribing to route params, so we can have switch perhaps

Reply:
Yeah it was already subscribing & new added param(`rgw_export_type`) is self-explanatory for create of rgw export, so it'd be better just add an extra if condition in this case to simplify